### PR TITLE
Improve error handling lesson

### LINKED ISSUE #799: Error Handling: Include guidance when to use which type of error handling
While the lesson is great at explaining the mechanism for error handling (including the {:ok, ...}, {:error, ...} convention), I miss some guidance on when to use which type of error handling. I would rephrase the second paragraph like this:

The general conventions for the types of error handling in Elixir go like this:
  * for errors that are part of the normal operation of a function (e.g. a user entered a wrong type of date), a function (example/1) returns {:ok, result} and {:error, reason} accordingly
  * for errors that are not part of normal operations (e.g. being unable to parse configuration data) you throw an exception

Usually you also add a second version of the function with a ! (example!/1) that returns the unwrapped result or raises an error. This lesson will focus on interacting with errors of the latter type.

### PR DESCRIPTION
**Pull request by brain-geek**
Based on text from @holsky 

Fixes #799

### CODE REVIEW

File: en/lessons/advanced/error-handling.md [hunk 1]
```diff
@@ -11,6 +11,17 @@ This lesson will focus on interacting with the latter.
 
 {% include toc.html %}
 
+## General conventions
+
+At the moment, Elixir community has come to some conventions regarding returning errors:
```
Comment (SophieDeBenedetto) [inline en/lessons/advanced/error-handling.md]: `At the moment, Elixir community has come to some conventions regarding returning errors:` -> `At the moment, the Elixir community has come to some conventions regarding returning errors:` (add the word `the` before `Elixir community`)

File: en/lessons/advanced/error-handling.md [hunk 2]
```diff
@@ -11,6 +11,17 @@ This lesson will focus on interacting with the latter.
...
+
+* For errors that are part of the regular operation of a function (e.g. a user entered a wrong type of date), a function returns {:ok, result} and {:error, reason} accordingly.
```
Comment (SophieDeBenedetto) [inline en/lessons/advanced/error-handling.md]: Wrap code in backticks, ex: `{:ok, result}`

File: en/lessons/advanced/error-handling.md [hunk 3]
```diff
@@ -11,6 +11,17 @@ This lesson will focus on interacting with the latter.
...
+* For errors that are not part of normal operations (e.g. being unable to parse configuration data) you throw an exception.
+
+We generally handle standard flow errors by [Pattern Matching](../basics/pattern-matching/), but in this lesson, we are focusing on the second case - on the exceptions.
+
+Usually, in public APIs, you can also find a second version of the function with a ! (example!/1) that returns the unwrapped result or raises an error.
```
Comment (SophieDeBenedetto) [inline en/lessons/advanced/error-handling.md]: This is kind of minor but maybe we can say `Often` instead of `Usually`. I'm not sure that I actually do see this pattern super often, although I am a fan of implementing it. Basically just not sure how much of a community-wide standard it is so want to use slightly less strong language. Let me know if you disagree!

**Comment by brain-geek** [comment]
@SophieDeBenedetto Thanks for the review! Please take a look once again!

**Comment by doomspork** [comment]
:boom: This is a fantastic addition :+1:

**Comment by geekyi** [comment]
The link gives a 404 error. 

This isn't the only case: #2252. How GitHub markdown generates the URL (relative to website) vs how the relative URL natively works in browsers is different.
